src/ring_theory/integral_closure.lean

```diff
@@ -127,70 +128,73 @@ begin
+  have : ∀ (jk : (↑(y.product y) : set (A × A))), jk.1.1 * jk.1.2 ∈ (span R ↑y : submodule R A),
   { intros jk,
-    let j : ↥(↑m : set A) := ⟨jk.1.1, (finset.mem_product.1 jk.2).1⟩,
-    let k : ↥(↑m : set A) := ⟨jk.1.2, (finset.mem_product.1 jk.2).2⟩,
-    have hj : j.1 ∈ (span R ↑m : submodule R A) := subset_span j.2,
-    have hk : k.1 ∈ (span R ↑m : submodule R A) := subset_span k.2,
-    revert hj hk, rw hm, exact @is_submonoid.mul_mem A _ M _ j.1 k.1 },
-  rw ← set.image_id ↑m at this,
+    let j : ↥(↑y : set A) := ⟨jk.1.1, (finset.mem_product.1 jk.2).1⟩,
+    let k : ↥(↑y : set A) := ⟨jk.1.2, (finset.mem_product.1 jk.2).2⟩,
+    have hj : j.1 ∈ (span R ↑y : submodule R A) := subset_span j.2,
+    have hk : k.1 ∈ (span R ↑y : submodule R A) := subset_span k.2,
+    revert hj hk, rw hy, exact @is_submonoid.mul_mem A _ S _ j.1 k.1 },
+  rw ← set.image_id ↑y at this,
   simp only [finsupp.mem_span_iff_total] at this,
-  choose lm hlm1 hlm2,
-  let S₀' : finset R := lx.frange ∪ finset.bind finset.univ (finsupp.frange ∘ lm),
+  choose ly hly1 hly2,
+  let S₀' : finset R := lx.frange ∪ finset.bind finset.univ (finsupp.frange ∘ ly),
   let S₀ : set R := ring.closure ↑S₀',
   refine is_integral_of_subring (ring.closure ↑S₀') _,
   letI : algebra S₀ (algebra.comap S₀ R A) := algebra.comap.algebra _ _ _,
-  letI : module S₀ (algebra.comap S₀ R A) := algebra.module,
-  have : (span S₀ (insert 1 (↑m:set A) : set (algebra.comap S₀ R A)) : submodule S₀ (algebra.comap S₀ R A)) =
-      (algebra.adjoin S₀ ((↑m : set A) : set (algebra.comap S₀ R A)) : subalgebra S₀ (algebra.comap S₀ R A)),
+  letI hmod : module S₀ (algebra.comap S₀ R A) := algebra.module,
+  have : (span S₀ (insert 1 (↑y:set A) : set (algebra.comap S₀ R A)) : submodule S₀ (algebra.comap S₀ R A)) =
+      (algebra.adjoin S₀ ((↑y : set A) : set (algebra.comap S₀ R A)) : subalgebra S₀ (algebra.comap S₀ R A)),
   { apply le_antisymm,
     { rw [span_le, set.insert_subset, mem_coe], split,
       change _ ∈ ring.closure _, exact is_submonoid.one_mem _, exact algebra.subset_adjoin },
     rw [algebra.adjoin_eq_span, span_le], intros r hr, refine monoid.in_closure.rec_on hr _ _ _,
     { intros r hr, exact subset_span (set.mem_insert_of_mem _ hr) },
     { exact subset_span (set.mem_insert _ _) },
     intros r1 r2 hr1 hr2 ih1 ih2,
-    rw ← set.image_id (insert _ ↑m) at ih1 ih2,
+    rw ← set.image_id (insert _ ↑y) at ih1 ih2,
     simp only [mem_coe, finsupp.mem_span_iff_total] at ih1 ih2,
     have ih1' := ih1, have ih2' := ih2,
     rcases ih1' with ⟨l1, hl1, rfl⟩, rcases ih2' with ⟨l2, hl2, rfl⟩,
     simp only [finsupp.total_apply, finsupp.sum_mul, finsupp.mul_sum, mem_coe],
     rw [finsupp.sum], refine sum_mem _ _, intros r2 hr2,
     rw [finsupp.sum], refine sum_mem _ _, intros r1 hr1,
     rw [algebra.mul_smul_comm, algebra.smul_mul_assoc],
-    letI : module ↥S₀ A := _inst_6, refine smul_mem _ _ (smul_mem _ _ _),
+    letI : module ↥S₀ A := ‹_›, refine smul_mem _ _ (smul_mem _ _ _),
```

Comment:
The `‹_›` is `hmod` right? It's easier for both the reader and Lean to name it directly.

Reply:
Done.